### ISSUE TITLE
Continuous status reporting

### ISSUE THREAD

**Issue by dl1com** [AllYarnsAreBeautiful/ayab-desktop#257]
To be able to show more information about the current machine status and for improved possibilities for debugging, the machine shall transmit some information about its current state.
Currently, this is only done in "Test Mode" (no knitting possible), so the continuous data transmission does not interfere with the knitting process.
But I guess it should be feasible to knit and send data in parallel.
To do so, the message handling on both sides (AYAB GUI and Arduino Firmware) has to be reviewed and optimized.

The information we want send to the computer is:
- Type of carriage (K, L, G)
- Direction of movement
- "Initialized?"-State
- Carriage position

Data we could try to send also:
- Value of hall sensor measurement, e.g. to be able to debug broken magnets or hall sensors.

**Comment by oskay** [comment]
I'd suggest that the carriage position indicate both row and column.

**Comment by chris007de** [comment]
Good point. Perhaps I would extend it to count also the rows since the last hall sensor pass, to take debugging of #161 into account.

**Comment by Adrienne200** [comment]
So far so good, this will be a very nice feature, but it is still very slow (tested with post61, for 0.95). Move the carriage across and then you have to wait about 8 seconds while the carriage display catches up with where it actually is. Then it finally beeps and clicks up to the next row. Could it check every second or two where the carriage is now and just jump there?
The carriage display isn't centered. Stop the carriage with the magnet centered, on or between L1 and R1, and wait for the display to catch up. It should be in the middle of the graphic, and it isn't.
Would be nice to have numbers on the graphic, or show the position number somehow. Maybe just when the carriage comes to rest.
A direction indicator would be nice, an arrow down below the row number.

**Comment by chris007de** [comment]
I would propose to leave it as it is for now and do improvements in the next version as this is not a core feature and we are running out of time this year.
The remaining question is whether to enable people to use it in the current state in 0.95 or to hide the checkbox for this mode for now.
What do you think?

**Comment by Adrienne200** [comment]
The Status tab is still visible but is always grey. Did you want to suppress it too?

**Comment by glindsey** [comment]
Just wanted to note: I'm trying to debug our current AYAB KH-910 installation, and this feature would be _really_ helpful for that, so I'd love to see it in the software eventually. :)

**Comment by glindsey** [comment]
Update, in case anybody cares: I figured out the problem with our installation. Since the EMSL AYAB interface is no longer available, I ordered the BOM parts, had boards printed, and hand-soldered one. Well, turns out I bought PCF8574**A** chips instead of PCF8574, so their I2C addresses were different (0x38/0x39 instead of 0x20/0x21). What a pain!

**Comment by t0mpr1c3** [comment]
What's the current status with this?

**Comment by chris007de** [comment]
Nothing has happened here since my last commit.

**Comment by t0mpr1c3** [comment]
I'm curious about the lag in serial communication. Do you have a sense of where it originates? If it is a buffering issue then a handshaking protocol might be the way to go.
